src/backend/src/controllers/projects.controllers.ts

```diff
@@ -1,341 +1,101 @@
+    try {
+      const wbsNumber: WbsNumber = validateWBS(req.params.wbsNum);
 
-  return res.status(200).json(projectTransformer(wbsEle));
-};
+      const project: Project = await ProjectsService.getSingleProject(wbsNumber);
 
-export const newProject = async (req: Request, res: Response) => {
-  // verify user is allowed to create projects
-  const user = await prisma.user.findUnique({ where: { userId: req.body.userId } });
-  if (!user) return res.status(404).json({ message: `user #${req.body.userId} not found!` });
-  if (user.role === Role.GUEST) return res.status(403).json({ message: 'Access Denied' });
-
-  await validateChangeRequestAccepted(req.body.crId);
-
-  if (req.body.teamId !== undefined) {
-    const team = await prisma.team.findUnique({ where: { teamId: req.body.teamId } });
-    if (!team) {
-      return res.status(404).json({ message: `team with id ${req.body.teamId} not found.` });
+      return res.status(200).json(project);
+    } catch (error: unknown) {
+      next(error);
     }
   }
 
-  // create the wbs element for the project and document the implemented change request
-  const maxProjectNumber = await getHighestProjectNumber(req.body.carNumber);
-  const createdProject = await prisma.wBS_Element.create({
-    data: {
-      carNumber: req.body.carNumber,
-      projectNumber: maxProjectNumber + 1,
-      workPackageNumber: 0,
-      name: req.body.name,
-      project: { create: { summary: req.body.summary, teamId: req.body.teamId } },
-      changes: {
-        create: {
-          changeRequestId: req.body.crId,
-          implementerId: req.body.userId,
-          detail: 'New Project Created'
-        }
-      }
-    },
-    include: { project: true, changes: true }
-  });
-
-  return res
-    .status(200)
-    .json(`${createdProject.carNumber}.${createdProject.projectNumber}.${createdProject.workPackageNumber}`);
-};
-
-export const editProject = async (req: Request, res: Response) => {
-  const { body } = req;
-  const { projectId, crId, userId, budget, summary, rules, goals, features, otherConstraints, name } = body;
+  static async newProject(req: Request, res: Response, next: NextFunction) {
+    try {
+      const user: User = await getCurrentUser(res);
+      const { crId, carNumber, name, summary, teamId } = req.body;
 
-  // Create optional arg values
-  const googleDriveFolderLink = body.googleDriveFolderLink === undefined ? null : body.googleDriveFolderLink;
-  const slideDeckLink = body.slideDeckLink === undefined ? null : body.slideDeckLink;
-  const bomLink = body.bomLink === undefined ? null : body.bomLink;
-  const taskListLink = body.taskListLink === undefined ? null : body.taskListLink;
-  const projectLead = body.projectLead === undefined ? null : body.projectLead;
-  const projectManager = body.projectManager === undefined ? null : body.projectManager;
+      const createdWbsNumber: WbsNumber = await ProjectsService.newProject(user, crId, carNumber, name, summary, teamId);
```

Comment:
On line 34 we might consider changing it to 'createdWbsNumber: string', which is what the createWorkPackage function returned.

Reply:
Nevermind, just realized wbsPipe might be doing that!

Reply:
yeah wbsPipe does that, see my other comment